mathesar/rpc/databases/configured.py

```diff
@@ -101,3 +107,5 @@ def disconnect(
         password=password,
     )
     database.delete()
+    if disconnect_db_server:
```

Comment:
This makes me a little nervous.

Consider a race condition where:

1. A `sales` DB is the only DB connected to Mathesar.
1. Alice loads the database page and opens the "Disconnect DB" modal. The modal renders a checkbox allowing Alice to delete the server.
1. Bob adds a `products` DB to Mathesar, associated with the same server.
1. Alice submits the modal form, deleting the server and the DB connection which Bob just added.

I'd prefer that `disconnect_db_server` have some safeguard logic which also checks to make sure that the database being disconnected is indeed the last database associated with the server.